### ISSUE TITLE
Update fog-openstack from 0.1.x to 0.2.x

### ISSUE THREAD

**Issue by chaadow** [chaadow/activestorage-openstack#10]
Right now, the gem has been properly tested with the latest 0.1 version of `fog-openstack` ( `0.1.27`)
Recently the `0.2.x` has been released, which could lead to breaking changes.

- [x] ~~Re-run the tests with default configuration~~
  - Configuration needs to be changed. Specifically the `openstack_auth_url` as explained in this [comment](https://github.com/chaadow/activestorage-openstack/issues/10#issuecomment-419712625)
  - tests pass with new configuration
- [x] ~~Wait for `fog-openstack` to merge @cyrylas PRs before considering updating the gem's version~~
  - [x] Fog-openstack has released version 0.2.2 on sep 11, which fixes all the `@host` issues. (ie. this [Pull Request](https://github.com/fog/fog-openstack/pull/424)
- [x] Update README configuration example for people migrating from 0.1 to 0.2 version.
  - As requested, fog-openstack updated the README accordingly, and this gem will help its users, as well as pointing to the Fog-openstack's README for more infos on how to setup the keys/configurations

**Comment by cyrylas** [comment]
Hi, I was just playing with version 0.2.x and openstack support is broken there.
I have filled two pull requests to fix that:
https://github.com/fog/fog-core/pull/243
https://github.com/fog/fog-openstack/pull/423

When they will be resolved, everything should work.

**Comment by chaadow** [comment]
Hi, thank you i was quickly trying the 0.2 version yesterday, trying to make it work. and the tests that were generating URLs were failing because the host was missing. The URLs generated were somewhat beginning like this : `https::443/`
Hopefully they'll review your PR.

**Comment by chaadow** [comment]
@cyrylas In the meantime, I managed to send the host to : 
```ruby
 def get_object_https_url(container, object, expires, options = {})
   create_temp_url(container, object, expires, "GET", {:port => 443}.merge(options).merge(:scheme => "https"))
 end
```
And make all the tests pass.  However I've made some changes to reflect on the latest API release changes : 
- In my configuration i've changed the `openstack_auth_uri` from : 
```yaml
openstack_auth_url: https://auth.example.com/v2.0/tokens
``` 
to : 
```yaml
openstack_auth_url: https://auth.example.com
```
( as mentionned in their commit, specify the version in the auth_url would break things.

- Second, to specify a version I've added this key to my configuration ( not necessary since it's the default version while exploring the `fog-openstack` but it's good to know: 
```yaml
openstack_identity_api_version: v2.0
```

- Finally while debugging i noticed that when I instantiate the client : 
```ruby
@client = Fog::Storage::OpenStack.new(settings)
```
I have access to the `:openstack_management_uri` instance variable, from which i can take the host :
```ruby
@client.instance_values["openstack_management_uri"].host 
```
 and send it as an  option param to `get_object_https_url` which chain it to `create_temp_url` and set the proper host like this : 
```ruby
host = options[:host] || @host
```

and in my gem I use it like this : 
```ruby
    def url(key, expires_in:, disposition:, filename:, **)
      instrument :url, key: key do |payload|
        expire_at = unix_timestamp_expires_at(expires_in)
        generated_url = client.get_object_https_url(container, key, expire_at, host: client.instance_values["openstack_management_uri"].host)
       ...
        payload[:url] = generated_url

        generated_url
      end
    end
```

We'll have to wait for your PRs, maybe the maintainers want people to manually send the host themselves.

**Comment by chaadow** [comment]
@cyrylas Let me know what you think about getting the host after instantiating the client and also what you think about the proposed solution

**Comment by cyrylas** [comment]
It's good to know, that we can workaround host problem. But I think its obvious mistake not populating `@host`. If the maintainers would like to send host manually they would not include `|| @host` in the code.

However second problem with string converted into numbers remain and I don't think it can solved on our side.
I've checked now, that fog-core was changing string into numbers like since forever. So probably older version of fog-openstack was converting tennant somewhere back to string. And new version don't.

**Comment by chaadow** [comment]
Fair enough, I'll wait a bit 👌

**Comment by chaadow** [comment]
@cyrylas They've just released version 0.2.2, thanks for letting them know
